New getting started for Angular (#2204)
aws-amplify/docs: docs/start/getting-started/fragments/angular/data-model.md

# Yes
```

> The schema generated is for a Todo app. Replace the GraphQL schema at `amplify/backend/api/RestaurantAPI/schema.graphql` with the following:

Reviewer: ```suggestion
The generated schema is for a Todo app. Replace the GraphQL schema at `amplify/backend/api/RestaurantAPI/schema.graphql` with the following:
```

```graphql
type Restaurant @model {
  id: ID!
  name: String!
  description: String!
  city: String!
}
```

>  You'll notice a directive on the `Restaurant` type of `@model`. This directive is part of the [GraphQL transform](~/cli/graphql-transformer/directives.md) library of Amplify. 

Reviewer: ```suggestion
 You'll notice a directive on the `Restaurant` type of `@model`. This directive is part of Amplify's [GraphQL transformer](~/cli/graphql-transformer/directives.md) functionality.
```

### Code generation

> Since you added an API the `amplify push` process will automatically prompt for codegen (select `yes` when prompted for automatic query and code generation). Choose **Angular** which will create an `API.service.ts` file in the app directory.

Reviewer: ```suggestion
Because you added an API, the `amplify push` process will automatically prompt for codegen. Select `yes` when prompted for automatic query and code generation. Choose **Angular**, which will create an `API.service.ts` file in the app directory.
```

In your `src/app/app.component.ts` file, add the following imports and modifications to your class to to add data to your database with a mutation by using the `API.service` file which was generated when you ran `amplify add api`:

```javascript
import { FormBuilder, FormGroup, Validators } from '@angular/forms';
import { APIService } from '../API.service';

@Component(...)
export class AppComponent implements OnInit {
  public createForm: FormGroup;

  constructor(private api: APIService, private fb: FormBuilder) { }

  async ngOnInit() {
    this.createForm = this.fb.group({
      'name': ['', Validators.required],
      'description': ['', Validators.required],
      'city': ['', Validators.required]
    });
  } 
  
  public onCreate(restaurant: any) {
    this.api.CreateRestaurant(restaurant).then(event => {
      console.log('item created!');
      this.createForm.reset();
    })
    .catch(e => {
      console.log('error creating restaurant...', e);
    });
  }
}
```

> Add the following to your `src/app/app.component.html` to add the form to create a new restaurant:

Reviewer: ```suggestion
Next, add a form that will be used for creating restaurants. Add the following to your `src/app/app.component.html`:
```

```html
<div class="form-body">
  <form autocomplete="off" [formGroup]="createForm" (ngSubmit)="onCreate(createForm.value)">
    <div>
      <label>Name: </label>
      <input type="text" formControlName="name" autocomplete="off">
    </div>
    <div>
      <label>Description: </label>
      <input type="text" formControlName="description" autocomplete="off">
    </div>
    <div>
      <label>City: </label>
      <input type="text" formControlName="city" autocomplete="off">
    </div>
    <button type="submit">Submit</button>
  </form>
</div>
```

> Create a file at `types/restaurant.ts` to define a type for the Restaurant:

Reviewer: ```suggestion
Now, define a `Restaurant` type. Create a new file at `types/restaurant.ts`:
```

```ts
export type Restaurant = {
  id : string,
  name : string,
  description : string,
  city: string
}; 
```

> Next, update the class to list all items in the database by running a `ListRestaurants` query when the app starts by implementing [OnInit](https://angular.io/api/core/OnInit) and storing the items in an array:

Reviewer: ```suggestion
Next, update your `AppComponent` class so that it will list all restaurants in the database when the app starts. To do so, implement [OnInit](https://angular.io/api/core/OnInit) add a `ListRestaurants` query. Store the query results in an array.
```

```javascript
import { APIService } from '../API.service';
import { Restaurant } from './../types/restaurant';

@Component(...)
export class AppComponent implements OnInit {
  restaurants: Array<Restaurant>;

  constructor(private api: APIService) { }

  ngOnInit() {
    this.api.ListRestaurants().then(event => {
      this.restaurants = event.items;
    });
  }
}
```

Add the following to your `src/app/app.component.html` to display any of the todos you have added:

```html
<div *ngFor="let restaurant of restaurants">
  <div>{{ restaurant.city }}</div>
  <div>{{ restaurant.name }}</div>
  <div>{{ restaurant.description }}</div>
</div>
```

> Finally, to subscribe to realtime data, update `ngOnInit` to setup a subscription on app start and update the `restaurants` array when new events are received:

Reviewer: ```suggestion
Finally, to subscribe to realtime data, update `ngOnInit`. When the app starts, setup a subscription. The subscription will update the `restaurants` array when new events are received:
```

```javascript
@Component(...)
export class AppComponent implements OnInit {
  ngOnInit() {
    //Subscribe to changes
    this.api.OnCreateRestaurantListener.subscribe( (event: any) => {
      const newRestaurant = event.value.data.onCreateRestaurant;
      this.restaurants = [newRestaurant, ...this.restaurants];
    });
  }
}
```

<amplify-callout>The code above imports only the API and PubSub category. To import the entire Amplify library use `import Amplify from 'aws-amplify'`. However, importing only the required categories is recommended as it will greatly reduce the final bundle size.
</amplify-callout>

> After restarting your app using `ng serve` go back to your browser and using dev tools you will see data being stored and retrieved in your backend from the console logs. 

Reviewer: ```suggestion
Restart your app by invoking `ng serve`. Now, go back to your browser. In the console logs, you will see data being stored and retrieved in your backend. 
```

^ I'm not an Angular guy, but the wording "dev tools," is that a term of art in the framework? If so, maybe "the Angular dev tools"?

Author: These refer to the browser dev tools so are not dependent on Angular.
